code/modules/mob/living/carbon/life.dm

```diff
@@ -314,7 +314,10 @@
 /mob/living/carbon/handle_status_effects()
 	..()
 	if(staminaloss)
-		adjustStaminaLoss(-3)
+		if(IsKnockdown() || IsUnconscious())

```

Comment:
Then you get the IsKnockdown checking for if staminaloss

Not really seeing how that would be better.